### PR TITLE
[Feat2-FR2] Restaurant Repo Update

### PR DESCRIPTION
**Pull request by Graysonzz**
## Description  
This PR implements the tagging fuctionality into the repository. The specific changes for this PR can be found in the restaurant repo file at line 56 (def add_menu_item) and its corresponding test is in test_restaurant_repo at line 56 (def test_add_menu_item_with_tags). Small changes were made to the conftest and menuItem model regarding tagging.

All other edits are from a required merge from PR #191 and PR #185 

## Changes
- [x] Adding feature to pre-existing file
- [x] Documentation

## Checklist
- [x] All tests pass
- [x] Test documation submitted to folder
- [x] Added necessary comments
- [x] Formatted correctly

### CODE REVIEW

**Review by bradyb2005 — COMMENTED** [review]
Take a look at my comment after you address that i'll run tests.

File: backend/models/user/restaurant_owner_model.py [hunk 1]
```diff
@@ -9,9 +9,6 @@
 class RestaurantOwner(User):
     res_id_counter = 1
```
[the remark this answers is not in the document]
Comment (Graysonzz) [reply backend/models/user/restaurant_owner_model.py]: Thanks for the question, no it is not, it has to explicitly call the method in order to create one. The counter starts at one and increments globally each time create_restaurant is called to ensure unique IDs!

**Review by fabihafifa — COMMENTED** [review]
I think it's great. No concerns. The restaurant repository change for adding menu items with tags makes sense, and the extra test for tagged menu items seems reasonable too.

**Review by aratnala7 — COMMENTED** [review]
looks good overall grayson, only thing like i mentioned, the new test covers the main happy path, but i think it would be nice to add edge case tests too, like missing restaurant id, empty tags, or multiple tags.

other than that, the implementation looks reasonable to me. 

im gonna run it and then get it approved.

File: backend/repositories/restaurant_repository.py [hunk 2]
```diff
@@ -52,7 +53,7 @@ def update_restaurant(self, restaurant: Restaurant) -> bool:
                 return True
         return False
 
-    def add_menu_item(self, restaurant_id: str, menu_item_data: Dict):
+    def add_menu_item(self, restaurant_id: str, menu_item: "MenuItem"):
```
[the remark this answers is not in the document]
Comment (Graysonzz) [reply backend/repositories/restaurant_repository.py]: Yes, thank you for mentioning this, it happens right below in the item_dict!
